src/crimson/os/seastore/cache.cc

```diff
@@ -1194,6 +1289,19 @@ void Cache::complete_commit(
 	    ? i->last_rewritten
 	    : seastar::lowres_system_clock::time_point());
       }
+      if (i->is_logical() || is_lba_node(i->get_type())) {
+	backref_list.emplace_back(
+	  std::make_unique<backref_buf_entry_t>(
+	    i->get_paddr(),
+	    i->is_logical()
+	    ? i->cast<LogicalCachedExtent>()->get_laddr()
+	    : i->cast<lba_manager::btree::LBANode>()->get_node_meta().begin,
+	    i->get_length(),
+	    i->get_type(),
+	    seq));
+      }
+      if (is_backref_node(i->get_type()))
+	add_backref_extent(i->get_paddr(), i->get_type());
```

Comment:
> Every extent is either a backref node or in the backref tree, right?

Um, RootBlock isn't neither of them.

Reply:
Our code should be self-documenting for easier review and also for future maintainances.

If the conditions are exclusive, we should always make this explict by using `if ... else if ... else ...` and assert out the impossible conditions if there are any. Otherwise we will tend to forget the invarients as the code evolves.

In this case, probably:
```c++
if (is_back_ref_node(...)) {
   ...
} else if (is_backref_mapped_extent_node(...)) 
   ...
}
```

Reply:
yes, will do